### DISCUSSION TITLE
Add Local Headphone Output as Player

### DISCUSSION THREAD

**Discussion by justynbell** [music-assistant/support#1157] [Feature requests and ideas]
So forgive me if:
1. This has already been discussed (I've been skimming, but it's late and I haven't dug into every discussion)
2. This is outside of the scope of MA, and more of an HA feature that MA could possibly take advantage of
3. It's a narrow use case

but
I have a HA Yellow sitting in my networking/AV rack right next to two Monoprice 6-zone amps.  It would be awesome if the local headphone jack could be a player so that the input to the amps could come straight from the Yellow.  Right now, I'm using a Raspberry Pi with piCorePlayer and the SlimProto integration in HA, but cutting out the extra HW, the channel between the two, and extra integration would be really cool for users whose Pis/Servers/Yellow that host HA are located physically close to their whole home audio setups.

If there's already a way to do this, I would love to know.

**Comment by marcelveldt** [comment]
@erkr you discovered some HA add-on right which hosts a local squeezelite player for exactly this use case ?

**Reply by erkr** [reply]
Yep:
https://github.com/pssc/ha-addon-squeezelite

**Reply by justynbell** [reply]
Oh sweet.  That's awesome.  In all my time searching - even outside the context of MA - the best I found for playing media locally was [this](https://community.home-assistant.io/t/required-a-step-by-step-guide-on-how-to-play-sounds-and-media-through-a-3-5mm-audio-jack/271478/5) post, which I didn't really like that solution.

I'll try it out!

**Reply by justynbell** [reply]
@erkr Since it sounds like you know a little bit about this, the only downside for the local Squeezelite player (which is actually a pretty decent solution, imo) is that you can't install the SlimProto integration in HA to also play announcements/media from HA because they both are listening on the same port.

I don't suppose you know of a configuration I'm missing, or is this just how it is?

**Reply by OzGav** [reply]
Currently this is how it is. Investigation into whether it is possible to make this configurable is ongoing.

**Reply by justynbell** [reply]
I know this is an old thread, but I erroneously thought the Squeezelite addon you linked and SlimProto integration were contending over that port, but I was wrong.  I only discovered tonight after playing with MA it's the SlimProto HA integration and *MA*.

Which for the record, I don't give a shit about the SlimProto HA integration given the progress on MA: this is going to be the one-stop-shop for all media on my HA server.  I won't have a need for the SlimProto integration.  Also, given the paradigm that "HA is not a media player", using MA and the squeezelite addon is actually a pretty good solution that applies to all setups.  Adding a headphone jack as a media player doesn't make much sense.

Super kudos on the progress of this platform.  It's really *really* good and it's getting better every day.

**Reply by marcelveldt** [reply]
Thanks!

**Comment by OzGav** [comment]
Local audio output has been added to 2.9.0beta4

**Reply by dfroelicher** [reply]
Great to hear. I downloaded the beta10 and can see it in the player providers page but I am not seeing it as an option when I go to play a track. Is there more setup that needs to be done. The documentation only lists squeezelite as a workflow. I also am not seeing it in the players page of the settings page.

Pi5 16gb USB headphone dongle
HA 2026.5.0

**Reply by dfroelicher** [reply]
It also seems to spin on playlists unable to play the playlist

**Reply by OzGav** [reply]
You need to look in MA SETTINGS >> PLAYERS to see if any players were created by the provider. If not, look in the logs.

**Reply by nTx-Cheerok** [reply]
I just tested it myself on my home assitant yellow and the logs just say.
2026-05-08 18:46:51.533 INFO (MainThread) [music_assistant] Loaded player provider Local Audio Out
2026-05-08 18:46:51.539 INFO (MainThread) [music_assistant.local_audio.bridge_manager] No local audio output devices found
2026-05-08 18:46:51.544 INFO (MainThread) [music_assistant.local_audio.bridge_manager] No local audio output devices found
2026-05-08 18:46:52.055 INFO (MainThread) [music_assistant] Loaded plugin provider Party
Do i need to enable like access to the onboard device ?

**Reply by dfroelicher** [reply]
> You need to look in MA SETTINGS >> PLAYERS to see if any players were created by the provider. If not, look in the logs.

Logs just say no local audio outputs found

**Reply by OzGav** [reply]
Did you read below? Does that apply to your situation?

**Reply by dfroelicher** [reply]
> Did you read below? Does that apply to your situation?

Below where, I am running on a Rpi5 using full HAOS and a usb to 3.5mm adapter. I do not want to use sendspin or LMS/squeezelite. I tried that path with no luck previously. I want this to be a fully self contained setup on one unit.

**Reply by OzGav** [reply]
Immediately below your post. Did you try the dev app to see if your USB device work?

**Reply by dfroelicher** [reply]
> Immediately below your post. Did you try the dev app to see if your USB device work?

What dev app, I am just getting into this side of things and am not familiar with that. How would that tell me if the USB is working if the program isn't finding it. I have the USB in the system area of HA and it was able to be selected in LMS and other third party apps when I was trying to go that route, so yes I believe the dongle works as it was working for a time and then wouldn't work anymore, probably after an update.

**Reply by OzGav** [reply]
Because we have enabled different build options on that. Sounds like you should just wait for future stable builds and see if it works then. As per below there is a good chance it will

**Reply by dfroelicher** [reply]
> Because we have enabled different build options on that. Sounds like you should just wait for future stable builds and see if it works then. As per below there is a good chance it will

So you are saying that the dev version of MA will have the ability to run the usb dongle?

**Reply by OzGav** [reply]
Yes. You will see that is exactly what is described below.

**Reply by dfroelicher** [reply]
> Yes. You will see that is exactly what is described below.

Ok, so i installed the dev version of MA and it found the dongle and is available in the configuration page of the addon and is I think visible in the players as a sendspin item but with no joy. no track will play from any of my playlists regardless of the number of tracks on said list even on the sendspin chrome web player.
[d5369777_music_assistant_dev_2026-05-13T04-03-55.674Z.log](https://github.com/user-attachments/files/27688319/d5369777_music_assistant_dev_2026-05-13T04-03-55.674Z.log)
 hope this logfile can shed some light.

**Reply by OzGav** [reply]
Spotify is failing in the log. Try a radio station

**Reply by dfroelicher** [reply]
> Spotify is failing in the log. Try a radio station

why would Spotify be failing, I need to use spotify not radio or other sources. its authenticated in providers settings. I am able to get the dongle to work with mother earth radio but spotify just brings back a "no playable item to start playback" message with a slew of error lines for all the tracks in the log. Really need spotify to work, without it the quest for local output is moot.

**Reply by OzGav** [reply]
Due to the Spotify hack earlier this year Spotify has stopped new accounts from working with 3rd party services. So if your account was created within the last six months or so then it unfortunately won’t work. You might like to switch to Tidal or Qobuz

**Reply by dfroelicher** [reply]
> Due to the Spotify hack earlier this year Spotify has stopped new accounts from working with 3rd party services. So if your account was created within the last six months or so then it unfortunately won’t work. You might like to switch to Tidal or Qobuz

Nope, had the account since like 2010.

**Reply by OzGav** [reply]
So playback to any other player works fine does it?

**Comment by OzGav** [comment]
OK the audio devices need to be passed through to the MA container. Also there are some build options we tweaked on the dev addon . If you want to to see if that fixes it then make sure your HA user has advanced mode turned on and then install the DEV version of the server as another app. Shutdown your usual MA server and start the dev version. It can take 10+ minutes to start. Once it does have a look in the log and see  if any devices are discovered now

**Reply by nTx-Cheerok** [reply]
Worked like a charm. Got added via Sendpsin. It shows up as audio device "pulse", is selectable and is usable as an output. 
Plays music just fine :)
2026-05-09 13:32:03.694 INFO (MainThread) [music_assistant] Loaded player provider Sendspin
2026-05-09 13:32:03.695 INFO (MainThread) [aiosendspin.server.server] Starting Sendspin server on port 8927
2026-05-09 13:32:03.723 INFO (MainThread) [aiosendspin.server.server] Sendspin server started successfully on 0.0.0.0:8927
2026-05-09 13:32:04.357 INFO (MainThread) [music_assistant] Loaded player provider Local Audio Out
2026-05-09 13:32:04.393 INFO (MainThread) [music_assistant.local_audio.bridge_manager] Found 2 local audio output device(s)
2026-05-09 13:32:04.403 INFO (MainThread) [music_assistant.players] Player (type player) registered: 0dcc4ef5-ed5c-55e9-a3bb-d742cfb7c19e/pulse
2026-05-09 13:32:04.414 INFO (MainThread) [music_assistant.Local Audio Out.bridge.pulse] Sendspin bridge registered for pulse (client_id=spb_0dcc4ef5ed5c55e9a3bbd742cfb7c19e)
2026-05-09 13:32:04.414 INFO (MainThread) [music_assistant.local_audio.bridge_manager] Bridge created for pulse
2026-05-09 13:32:04.465 INFO (MainThread) [music_assistant.players] Player (type protocol) registered: spb_0dcc4ef5ed5c55e9a3bbd742cfb7c19e/pulse
2026-05-09 13:32:04.487 INFO (MainThread) [music_assistant.players] Player (type player) registered: 17f2a88d-b60b-55e8-bf8f-1e833685f390/default
2026-05-09 13:32:04.556 INFO (MainThread) [music_assistant.Local Audio Out.bridge.default] Sendspin bridge registered for default (client_id=spb_17f2a88db60b55e8bf8f1e833685f390)
2026-05-09 13:32:04.557 INFO (MainThread) [music_assistant.local_audio.bridge_manager] Bridge created for default
2026-05-09 13:32:04.576 INFO (MainThread) [music_assistant.players] Player (type protocol) registered: spb_17f2a88db60b55e8bf8f1e833685f390/default
2026-05-09 13:32:04.581 INFO (MainThread) [music_assistant.local_audio.bridge_manager] Found 2 local audio output device(s)
2026-05-09 13:32:12.241 INFO (MainThread) [music_assistant.music.genre] Genre mapping scan completed: 0 items mapped (0.5s)
2026-05-09 13:32:18.150 INFO (MainThread) [music_assistant] Loaded plugin provider Party
2026-05-09 13:32:18.192 INFO (MainThread) [music_assistant] Loaded player provider Chromecast
2026-05-09 13:32:18.231 INFO (MainThread) [music_assistant] Loaded player provider DLNA

**Reply by OzGav** [reply]
OK I need to roll out those changes to the ~~dev~~ beta and stable servers. We will get that done soon.

**Reply by nTx-Cheerok** [reply]
Amazing, thanks for the quick action. Love this addon!

**Reply by justynbell** [reply]
I can't wait to try this on my Yellow.  It's a shame they discontinued that hardware, but it should support [higher quality audio](https://community.home-assistant.io/t/the-yellows-headphone-jack-should-be-a-first-class-citizen/550783/8?u=justynnuff) than a normal Pi, so I'm looking forward to seeing if/how this works with the MA dev addon.  If it does work, you've essentially provided a solution to my [request ](https://community.home-assistant.io/t/the-yellows-headphone-jack-should-be-a-first-class-citizen/550783)from 3 years ago :)

**Reply by KipiKland** [reply]
I have a HA Yellow connected through USB to a Topping D50 DAC, i'd love to use MA directly with the DAC <3.

BUSNUM: '001'
DEVNAME: /dev/bus/usb/001/003
DEVNUM: '003'
DEVPATH: /devices/platform/soc/fe980000.usb/usb1/1-1/1-1.1
DEVTYPE: usb_device
DRIVER: usb
ID_BUS: usb
ID_MODEL: D50s
ID_MODEL_ENC: D50s
ID_MODEL_ID: '8750'
ID_PATH: platform-fe980000.usb-usb-0:1.1
ID_PATH_TAG: platform-fe980000_usb-usb-0_1_1
ID_PATH_WITH_USB_REVISION: platform-fe980000.usb-usbv2-0:1.1
ID_REVISION: '0103'
ID_SERIAL: Topping_D50s
ID_USB_INTERFACES: ':010120:010220:fe0101:'
ID_USB_MODEL: D50s
ID_USB_MODEL_ENC: D50s
ID_USB_MODEL_ID: '8750'
ID_USB_REVISION: '0103'
ID_USB_SERIAL: Topping_D50s
ID_USB_VENDOR: Topping
ID_USB_VENDOR_ENC: Topping
ID_USB_VENDOR_ID: 152a
ID_VENDOR: Topping
ID_VENDOR_ENC: Topping
ID_VENDOR_ID: 152a
MAJOR: '189'
MINOR: '2'
PRODUCT: 152a/8750/103
SUBSYSTEM: usb
TYPE: 239/2/1
USEC_INITIALIZED: '5054136'



DEVLINKS: >-
  /dev/snd/by-id/usb-Topping_D50s-00
  /dev/snd/by-path/platform-fe980000.usb-usb-0:1.1:1.0
  /dev/snd/by-path/platform-fe980000.usb-usbv2-0:1.1:1.0
DEVNAME: /dev/snd/controlC1
DEVPATH: >-
  /devices/platform/soc/fe980000.usb/usb1/1-1/1-1.1/1-1.1:1.0/sound/card1/controlC1
ID_BUS: usb
ID_MODEL: D50s
ID_MODEL_ENC: D50s
ID_MODEL_ID: '8750'
ID_PATH: platform-fe980000.usb-usb-0:1.1:1.0
ID_PATH_TAG: platform-fe980000_usb-usb-0_1_1_1_0
ID_PATH_WITH_USB_REVISION: platform-fe980000.usb-usbv2-0:1.1:1.0
ID_REVISION: '0103'
ID_SERIAL: Topping_D50s
ID_TYPE: audio
ID_USB_DRIVER: snd-usb-audio
ID_USB_INTERFACES: ':010120:010220:fe0101:'
ID_USB_INTERFACE_NUM: '00'
ID_USB_MODEL: D50s
ID_USB_MODEL_ENC: D50s
ID_USB_MODEL_ID: '8750'
ID_USB_REVISION: '0103'
ID_USB_SERIAL: Topping_D50s
ID_USB_TYPE: audio
ID_USB_VENDOR: Topping
ID_USB_VENDOR_ENC: Topping
ID_USB_VENDOR_ID: 152a
ID_VENDOR: Topping
ID_VENDOR_ENC: Topping
ID_VENDOR_ID: 152a
MAJOR: '116'
MINOR: '5'
SUBSYSTEM: sound
SYSTEMD_USER_WANTS: sound.target
SYSTEMD_WANTS: sound.target
TAGS: ':systemd:'
USEC_INITIALIZED: '6242993'
